File: 2016/03.rb
#!/usr/bin/env ruby

class Array
  def is_tri?
    a, b, c = self.sort
    a + b > c
  end
end

lines = File.open(ARGV.first)
            .readlines
            .map(&:chomp)

nums = lines.map { |l| l.split(" ").map(&:to_i) }

puts nums.count(&:is_tri?)

puts (0..nums[0].size - 1).sum { |col|
  c_nums = nums.collect { |n| n[col] }.each_slice(3)
  c_nums.count(&:is_tri?)
}

File: 2016/03.py
#!/usr/bin/env python3

import re
from itertools import islice


def isTri(arr):
    tmp = arr.copy()
    tmp.sort()
    a, b, c = tmp
    return a + b > c


nums = []
for line in open("input/03.txt").readlines():
    nums.append([int(x) for x in re.split('\s+', line.strip())])

print(sum((1 for x in nums if isTri(x))))

part2 = 0
for i in range(len(nums[0])):
    col = [row[i] for row in nums]
    for j in range(0, len(col), 3):
        if isTri(col[j: j+3]):
            part2 += 1

print(part2)
